Inc/PedalBox.h

```diff
@@ -8,32 +8,111 @@
 #ifndef PEDALBOX_H_
 #define PEDALBOX_H_
 
+#include <stdbool.h>
+#include "main.h"
+#include "cmsis_os.h"
 #include "stm32f4xx_hal.h"
 
+#define THROTTLE_SENSOR_PLAUS_THRESH  0.1f
+#define BRAKE_PRESSED_THRESHOLD       0.30f
+#define APPS_BP_PLAUS_RESET_THRESHOLD 0.05f
+#define APPS_BP_PLAUS_THRESHOLD       0.25f
+#define THROTTLE_LOWER_BOUND          0.15f
+
+//pedalbox defines //todo not sure if better to send whole frame or just pbmsg.
+#define PEDALBOX1_FILTER                0 //filter number corresponding to the PEDALBOX1 message
+#define PEDALBOX1_THROT1_7_0_BYTE       1
+#define PEDALBOX1_THROT1_7_0_OFFSET     0
+#define PEDALBOX1_THROT1_7_0_MASK       0b11111111
+#define PEDALBOX1_THROT1_11_8_BYTE      0
+#define PEDALBOX1_THROT1_11_8_OFFSET    0
+#define PEDALBOX1_THROT1_11_8_MASK      0b00001111
+#define PEDALBOX1_THROT2_7_0_BYTE       3
+#define PEDALBOX1_THROT2_7_0_OFFSET     0
+#define PEDALBOX1_THROT2_7_0_MASK       0b11111111
+#define PEDALBOX1_THROT2_11_8_BYTE      2
+#define PEDALBOX1_THROT2_11_8_OFFSET    0
+#define PEDALBOX1_THROT2_11_8_MASK      0b00001111
+//brake
+#define PEDALBOX1_BRAKE1_7_0_BYTE       5
+#define PEDALBOX1_BRAKE1_7_0_OFFSET     0
+#define PEDALBOX1_BRAKE1_7_0_MASK       0b11111111
+#define PEDALBOX1_BRAKE1_11_8_BYTE      4
+#define PEDALBOX1_BRAKE1_11_8_OFFSET    0
+#define PEDALBOX1_BRAKE1_11_8_MASK      0b00001111
+#define PEDALBOX1_BRAKE2_7_0_BYTE       7
+#define PEDALBOX1_BRAKE2_7_0_OFFSET     0
+#define PEDALBOX1_BRAKE2_7_0_MASK       0b11111111
+#define PEDALBOX1_BRAKE2_11_8_BYTE      6
+#define PEDALBOX1_BRAKE2_11_8_OFFSET    0
+#define PEDALBOX1_BRAKE2_11_8_MASK      0b00001111
+
+#define PEDALBOX1_EOR_BYTE            3
+#define PEDALBOX1_EOR_OFFSET          0
+#define PEDALBOX1_EOR_MASK            0b00000001
+#define PEDALBOX1_IMP_BYTE            3
+#define PEDALBOX1_IMP_OFFSET          1
+#define PEDALBOX1_IMP_MASK            0b00000010
+
+
+typedef enum 
+{
+  THROTTLE_1_MIN = 0x0FFF,
+  THROTTLE_1_MAX = 0x0490,
+  
+  THROTTLE_2_MIN = 0x0EA0,
+  THROTTLE_2_MAX = 0x0320,
+  
+} Throttle_Thresholds_t;
+
+typedef enum
+{
+  BRAKE_1_MIN = 0x027C,
+  BRAKE_1_MAX = 0x0900,
+
+  BRAKE_2_MIN = 0x026F,
+  BRAKE_2_MAX = 0x0900,
+} Brake_Thresholds_t;
```

Comment:
Brakes and throttle would be great values to store in eeprom. 

You could also potentially save config files on PI and transfer them over on boot but that seems more hacky.